### PR TITLE
3.5 index backed order by

### PR DESCRIPTION
**Pull request by craigtaverner**
Documentation for the performance optimization provided by index-backed property lookup and index-backed order-by.

### CODE REVIEW

File: cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc [hunk 1]
```diff
@@ -33,3 +36,77 @@ MERGE (p:Person {name: line.name})
 ON CREATE SET p.born = toInteger(line.born)
 MERGE (p)-[:DIRECTED]->(m)
 ----
+
+[source, cypher]
+----
+CREATE INDEX ON :Person(name)
```
Comment (jexp) [inline cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc:42]: add `call dbms.awaitIndexes(timeout)`?

Comment (craigtaverner) [reply cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc:42]: That did not help. I actually had to use a `USING INDEX` hint to get the queries to work.

File: cypher/cypher-docs/src/docs/dev/query-tuning.asciidoc [hunk 2]
```diff
@@ -148,3 +150,25 @@ The same goes for labels, where declaring labels on both the start and end nodes
...
+
+[abstract]
+--
+This section describes some more subtle optimizations based on new native index capabilities
+--
+
+One of the most important and useful ways of optimizing Cypher queries involves creating appropriate indexes.
+This is described in more detail in the section on '<<query-schema-index,Indexes>>', and demonstrated in the <<cypherdoc-basic-query-tuning-example,basic query tuning example>>.
+In summary an index will be based on the combination of a `Label` and a `property`.
+Any Cypher query that searches for Nodes with a specific label and some predicate on the property (equality, range or existence) will be planned to use
```
Comment (petraselmer) [inline cypher/cypher-docs/src/docs/dev/query-tuning.asciidoc]: I wouldn't title-case Nodes...

**Review by petraselmer — CHANGES_REQUESTED** [review]
@craigtaverner can you also add a 3.5 section to this file: https://github.com/neo4j/neo4j-documentation/blob/3.5/cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc  (and add a line about the IBO)

Thanks!

**Comment by petraselmer** [comment]
👍

**Review by petraselmer — APPROVED** [review]

**Pull request merged**